libmultilabel/model.py

```diff
@@ -35,13 +36,36 @@ def __init__(
         self.optimizer = optimizer
         self.momentum = momentum
         self.weight_decay = weight_decay
-        # evaluator
-        self.eval_metric = MultiLabelMetrics(metric_threshold, monitor_metrics)
+
         # dump log
         self.log_path = log_path
         self.silent = silent
         self.save_k_predictions = save_k_predictions
 
+        macro_prec = Precision(self.num_classes, metric_threshold, average='macro')
+        macro_recall = Recall(self.num_classes, metric_threshold, average='macro')
+        another_macro_f1 = 2 * (macro_prec * macro_recall) / (macro_prec + macro_recall + 1e-10)
+        metrics = {
+            'Micro-Precision': Precision(self.num_classes, metric_threshold, average='micro'),
+            'Micro-Recall': Recall(self.num_classes, metric_threshold, average='micro'),
+            'Micro-F1': F1(self.num_classes, metric_threshold, average='micro'),
+            'Macro-F1': F1(self.num_classes, metric_threshold, average='macro'),
+            # The f1 value of macro_precision and macro_recall. This variant of
+            # macro_f1 is less preferred but is used in some works. Please
+            # refer to Opitz et al. 2019 [https://arxiv.org/pdf/1911.03347.pdf]
+            'Another-Macro-F1': another_macro_f1,
+        }
+        for metric in monitor_metrics:
+            if re.match('P@\d+', metric):
+                metrics[metric] = Precision(self.num_classes, metric_threshold,
+                                            average='samples', top_k=int(metric[2:]))
+            elif re.match('R@\d+', metric):
+                metrics[metric] = Recall(self.num_classes, metric_threshold,
+                                         average='samples', top_k=int(metric[2:]))
+            elif metric not in ['Micro-Precision', 'Micro-Recall', 'Micro-F1', 'Macro-F1', 'Another-Macro-F1']:
+                raise ValueError(f'Invalid metric: {metric}')
+        self.eval_metric = MetricCollection(metrics)
+
```

Comment:
Maybe we can make  collection of metrics a class and move to metrics.py

Reply:
As `MetricCollection` is already a class of a collection of metrics, maybe we can use a function such as `get_metrics()` in `,metric.py` to get the collection?

Reply:
Sure make it a function helps too